var/spack/repos/builtin/packages/py-basemap/package.py

```diff
@@ -9,33 +9,37 @@ class PyBasemap(PythonPackage):
     """The matplotlib basemap toolkit is a library for plotting
     2D data on maps in Python."""
 
-    url = "https://github.com/matplotlib/basemap/archive/v1.2.0rel.tar.gz"
     homepage = "https://matplotlib.org/basemap/"
+    pypi = "basemap/basemap-1.4.1.zip"
 
     license("MIT")
 
-    version("1.2.1", sha256="3fb30424f18cd4ffd505e30fd9c810ae81b999bb92f950c76553e1abc081faa7")
+    version("1.4.1", sha256="730b1e2ff5eb31c73680bd8ebabc6b11adfc587cfa6832c528a8a82822e5a490")
+
+    variant("hires", default=False, description="Install hi-res data.")
 
     depends_on("c", type="build")  # generated
     depends_on("cxx", type="build")  # generated
 
+    depends_on("python@3.10:3.11", type=("build", "run"))
     # Per Github issue #3813, setuptools is required at runtime in order
     # to make mpl_toolkits a namespace package that can span multiple
     # directories (i.e., matplotlib and basemap)
     depends_on("py-setuptools", type=("build", "run"))
     depends_on("py-numpy@1.2.1:", type=("build", "run"))
-    depends_on("py-matplotlib@1.0.0:3.0.0,3.0.2:", type=("build", "run"))
+    depends_on("py-matplotlib", type=("build", "run"))
     # 1.2.1 is PROJ6 compatible
     # https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=939022
-    depends_on("py-pyproj@1.9.3:", type=("build", "run"), when="@1.2.1:")
+    depends_on("py-pyproj", type=("build", "run"))
```

Comment:
This should also be constrained per `requirements.txt`:

```
pyproj >= 1.9.3, < 2.1.0; python_version == "2.6"
pyproj >= 1.9.3, < 2.2.0; python_version == "2.7"
pyproj >= 1.9.3, < 1.9.6; python_version == "3.2"
pyproj >= 1.9.3, < 2.1.0; python_version == "3.3"
pyproj >= 1.9.3, < 2.1.0; python_version == "3.4"
pyproj >= 1.9.3, < 3.7.0; python_version >= "3.5"
```

```suggestion
    depends_on("py-pyproj@1.9.3:3.6", type=("build", "run"))
```

Reply:
I'll change it